gc/base/standard/Scavenger.cpp

```diff
@@ -1597,6 +1597,12 @@ MM_Scavenger::copy(MM_EnvironmentStandard *env, MM_ForwardedHeader* forwardedHea
 			_extensions->objectModel.fixupForwardedObject(forwardedHeader, destinationObjectPtr, objectAge);
 		}
 
+#if defined(OMR_ENV_DATA64)
+		if(_extensions->objectModel.isIndexable(destinationObjectPtr)) {
+			updateProxyObjectInternalDataAddr(destinationObjectPtr);
+		}
+#endif /* defined(OMR_ENV_DATA64) */
```

Comment:
I know I have suggested this is good spot in previous conversation, but for CS, this must go before commitFixup. The object has to completely valid (hence dataAddr updated), before we announce it in the the world (mutator threads).

Since nobody is relying on it being updated, it's not obvious to see the problem.

You can try to put it inside ObjectModel::fixupForwardedObject, although I don't know if indexableObjectModel is visible inside.

Reply:
I see, and you're right `indexableObjectModel` is not visible inside `ObjectModel`, I could create a helper inside `Scavenger.cpp` say `updateForwardedObject()` and in there I would have both `fixupForwardedObject` and `updateProxyObjectInternalDataAddr`